### PR TITLE
frontend/fix: #1123 added bonus in driver app

### PR DESCRIPTION
**Pull request by aryanguptaSG**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added Namma Bonus in Driver App . 
Moved `RequestInfoCard` component to common .
AssetStore PR :- https://bitbucket.juspay.net/projects/PICAF/repos/assetstore/pull-requests/264/overview
Output :- [IMG1](https://drive.google.com/file/d/1NNR-YfeQLLhrmc7MisYA8hB_Vk6T0v7v/view?usp=sharing) 
[IMG2](https://drive.google.com/file/d/12sMUR3CzvJAoXkSEmzaIqXCuMHtUt_4M/view?usp=sharing)

figma :- [FIGMA LINK](https://www.figma.com/file/4zT4QhImxqphGf56DpAjcV/Beckn-Release-Version?type=design&node-id=59172%3A12768&t=aZ8VUvwWnGfSR19X-1)

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 1]
```diff
@@ -658,7 +658,7 @@ requestInfoCardView push state =
         [ height MATCH_PARENT
         , width MATCH_PARENT
         ]
-        [ RequestInfoCard.view (push <<< RequestInfoCardAction)  ]
+        [ RequestInfoCard.view (push <<< RequestInfoCardAction) (requestInfoCardConfig "dummyString") ]
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:661]: @aryanguptaSG  Please add the Function of LazyCheck here
Don't use dummy strings

File: Frontend/ui-driver/src/Screens/HomeScreen/View.purs [hunk 2]
```diff
@@ -1006,6 +1008,15 @@ dummyTextView =
...
   ]
 
+requestInfoCardView :: forall w. (Action -> Effect Unit) -> HomeScreenState -> PrestoDOM (Effect Unit) w
+requestInfoCardView push state =
+  PrestoAnim.animationSet [ Anim.fadeIn true ]
+    $ linearLayout
+        [ height MATCH_PARENT
+        , width MATCH_PARENT
+        ]
+        [ RequestInfoCard.view (push <<< RequestInfoCardAction) (requestInfoCardConfig "dummyString") ]
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/HomeScreen/View.purs:1018]: Use the LazyCheck type

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by Rohit4997 — APPROVED** [review]

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 3]
```diff
@@ -167,6 +168,7 @@ instance loggableAction :: Loggable Action where
     TriggerMaps -> trackAppActionClick appId (getScreen HOME_SCREEN) "in_screen" "trigger_maps"
     RemoveGenderBanner -> trackAppActionClick appId (getScreen HOME_SCREEN) "in_screen" "gender_banner"
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:170]: @aryanguptaSG can you remove extra line

File: Frontend/ui-driver/src/Strings/BN.js [hunk 4]
```diff
@@ -247,6 +247,11 @@ const bengaliStrings = {
 	UPDATED_AT: "ম্যাপ শেষ আপডেট",
 	TRIP_COUNT: "আজকের ট্রিপস",
 	TODAYS_EARNINGS: "আজকের উপার্জন",
+	BONUS_EARNED : "বোনাস অর্জিত",
+	GOT_IT : "বুঝেছি!",
+	WHAT_IS_NAMMA_YATRI_BONUS : "নম্মা যাত্রী বোনাস কি?",
+	BONUS_PRIMARY_TEXT : "নম্মা যাত্রী বোনাস হল অতিরিক্ত পরিমাণ যা আপনি মিটার চার্জের উপরে পিকআপ চার্জ, গ্রাহকের টিপস এবং ড্রাইভার যোগের আকারে অর্জন করেছেন।",
+    BONUS_SECONDARY_TEXT : "নম্মা যাত্রী বোনাসের পরিমাণ আপনার মোট উপার্জনের অংশ।",
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Strings/BN.js:254]: @aryanguptaSG please try to maintain indentation.

File: Frontend/ui-driver/src/Strings/KN.js [hunk 5]
```diff
@@ -253,6 +253,11 @@ const kannadaStrings = {
   SELECT_DATE_OF_REGISTRATION: "ನೋಂದಣಿ ದಿನಾಂಕವನ್ನು ಆಯ್ಕೆಮಾಡಿ",
   TRIP_COUNT: "ಇಂದಿನ ಪ್ರವಾಸಗಳು",
   TODAYS_EARNINGS: "ಇಂದಿನ ಗಳಿಕೆ",
+  BONUS_EARNED : "ಬೋನಸ್ ಗಳಿಸಲಾಗಿದೆ",
+  GOT_IT : "ಅರ್ಥವಾಯಿತು!",
+  WHAT_IS_NAMMA_YATRI_BONUS : "ನಮ್ಮ ಯಾತ್ರಿ ಬೋನಸ್ ಎಂದರೇನು?",
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Strings/KN.js:258]: @aryanguptaSG please add translations in yatriConfig.js and jatriSathiConfig.js

**Review by MercyQueen — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Components/StatsModel/View.purs [hunk 6]
```diff
@@ -54,53 +54,86 @@ earningsView config push =
...
-            ] <> FontStyle.h2 TypoGraphy
-         ]
+  ][  linearLayout
+      [ width WRAP_CONTENT
+      , height WRAP_CONTENT
+      , orientation VERTICAL
+      , gravity CENTER
+      , weight 1.0
+      ][ textView $ 
+         [ height config.countTextConfig.height
```
Comment (aryan-13) [inline Frontend/ui-driver/src/Components/StatsModel/View.purs:64]: Make a common textview for all the textviews that are being used in this layout.

**Review by aryan-13 — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Strings/BN.js (hunk 4 above)
Comment (aryanguptaSG) [reply Frontend/ui-driver/src/Strings/BN.js:254]: it is already indented perfectly don't know why it is looking like this here .

**Review by aryan-13 — APPROVED** [review]

**Review by MercyQueen — APPROVED** [review]

**Review by kavya-shree-s — APPROVED** [review]

**Comment by kavya-shree-s** [comment]
@aryanguptaSG Have you tested customer app after this change

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**